subprojects/ivy/src/main/java/org/gradle/api/publish/ivy/tasks/PublishToIvyRepository.java

```diff
@@ -40,8 +44,12 @@
  */
 public class PublishToIvyRepository extends DefaultTask {
 
+    private static final Credentials NO_CREDENTIALS = new Credentials() {
```

Comment:
Instead of a constant value to represent a missing value, I would instead use an actual missing value and mark the `getCredentials()` property with `@Optional`.

Reply:
Thanks, this worked nicely: https://github.com/gradle/gradle/pull/13587